File: 2026-08/src/day_21/prms_42576/Solution.java
package day_21.prms_42576;

import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> cnt = new HashMap<>();

        for (String p : participant) {
            cnt.put(p, cnt.getOrDefault(p, 0) + 1);
        }

        for (String c : completion) {
            cnt.put(c, cnt.getOrDefault(c, 0) - 1);
        }

        for (Map.Entry<String, Integer> e : cnt.entrySet()) {
            if (e.getValue() == 1) return e.getKey();
        }

        return null;
    }
}

File: 2026-08/src/day_21/prms_42576/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map<string, int> cnt;
    for (string& p : participant) cnt[p]++;
    for (string& c : completion) cnt[c]--;

    for (auto& [k, v] : cnt) {
        if (v == 1) return k;
    }

    return "";
}
